src/main/java/frc/robot/subsystems/Climber.java

```diff
@@ -51,4 +115,55 @@ public void getElevatorHeight(){
   public void getArmAngle(){
 
   }
-}
+
+  private void checkArmPIDVal(){
+   // read PID coefficients from SmartDashboard
+   double p = SmartDashboard.getNumber("P Gain", 0);
+   double i = SmartDashboard.getNumber("I Gain", 0);
+   double d = SmartDashboard.getNumber("D Gain", 0);
+   double iz = SmartDashboard.getNumber("I Zone", 0);
+   double ff = SmartDashboard.getNumber("Feed Forward", 0);
+   double max = SmartDashboard.getNumber("Max Output", 0);
+   double min = SmartDashboard.getNumber("Min Output", 0);
+   double maxV = SmartDashboard.getNumber("Max Velocity", 0);
+   double minV = SmartDashboard.getNumber("Min Velocity", 0);
+   double maxA = SmartDashboard.getNumber("Max Acceleration", 0);
+   double allE = SmartDashboard.getNumber("Allowed Closed Loop Error", 0);
+
+   // if PID coefficients on SmartDashboard have changed, write new values to controller
+   if((p != m_kPArm)) { m_armPIDController.setP(p); m_kPArm = p; }
+   if((i != m_kIArm)) { m_armPIDController.setI(i); m_kIArm = i; }
+   if((d != m_kDArm)) { m_armPIDController.setD(d); m_kDArm = d; }
+   if((iz != m_kIzArm)) { m_armPIDController.setIZone(iz); m_kIzArm = iz; }
+   if((ff != m_kFFArm)) { m_armPIDController.setFF(ff); m_kFFArm = ff; }
+   if((max != m_kMaxOutputArm) || (min != m_kMinOutputArm)) { 
+     m_armPIDController.setOutputRange(min, max); 
+     m_kMinOutputArm = min; m_kMaxOutputArm = max; 
+   }
+   if((maxV != m_armMaxVel)) { m_armPIDController.setSmartMotionMaxVelocity(maxV,0); m_armMaxVel = maxV; }
+   if((minV != m_armMinVel)) { m_armPIDController.setSmartMotionMinOutputVelocity(minV,0); m_armMinVel = minV; }
+   if((maxA != m_armMaxAcc)) { m_armPIDController.setSmartMotionMaxAccel(maxA,0); m_armMaxAcc = maxA; }
+   if((allE != m_armAllowedErr)) { m_armPIDController.setSmartMotionAllowedClosedLoopError(allE,0); m_armAllowedErr = allE; }
+
+   double setPoint, processVariable;
+   boolean mode = SmartDashboard.getBoolean("Mode", false);
+   if(mode) {
+     setPoint = SmartDashboard.getNumber("Set Velocity", 0);
+     m_armPIDController.setReference(setPoint, CANSparkMax.ControlType.kVelocity);
+     processVariable = m_armEncoder.getVelocity();
+   } else {
+     setPoint = SmartDashboard.getNumber("Set Position", 0);
+     /**
+      * As with other PID modes, Smart Motion is set by calling the
+      * setReference method on an existing pid object and setting
+      * the control type to kSmartMotion
+      */
+     m_armPIDController.setReference(setPoint, CANSparkMax.ControlType.kSmartMotion);
```

Comment:
Add a TODO here that this should be moved to the setArmAngle method. Note that the rotateArm method should be changed to setArmAngle.